reactExercise/solutions/zihe/ex5/src/Lifecycle.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+class Lifecycle extends React.Component {
+    constructor() {
+        super()
+        this.state = {
+            x1: false,
+            x2: false,
+            x3: false,
+            loading: true,
+        }
+    }
+    componentDidMount() {
+        setTimeout(() => {
+            this.setState({ loading: false })
+        }, 3000)
+    }
+    shouldComponentUpdate(nextProps, nextState) {
+        if (nextState.x3 === true && nextState.x1 === false && nextState.x2 === false) {
+            this.setState({ x3: false })
+            return 0;
```

Comment:
虽然JavaScript会自动帮我们evaluate 0成false  但是直接写false会好一些
```suggestion
            return false;
```